map/framework.cpp

```diff
@@ -2980,3 +2982,26 @@ bool Framework::OriginalFeatureHasDefaultName(FeatureID const & fid) const
 {
   return osm::Editor::Instance().OriginalFeatureHasDefaultName(fid);
 }
+
+bool Framework::HasRouteAltitude() const { return m_routingSession.HasRouteAltitude(); }
+
+bool Framework::GenerateRouteAltitudeChart(size_t width, size_t height, bool day,
+                                           vector<uint8_t> & imageRGBAData) const
+{
+  feature::TAltitudes altitudes;
+  if (!m_routingSession.GetRouteAltitudes(altitudes))
+    return false;
+  deque<double> segDistanceM;
+  if (!m_routingSession.GetSegDistanceM(segDistanceM) && segDistanceM.empty())
+    return false;
+  segDistanceM.push_front(0.0);
```

Comment:
> No need in deque. Use vector::insert to the front() here.

Probably. It's not a big deal to copy the vector.

>  Amortized complexity O(1).

Are you sure? It's from C++11 standard:
"
...
23.3.6.5 vector modifiers [vector.modifiers]
iterator insert(const_iterator position, const T& x);
.....
Complexity: The complexity is linear in the number of elements inserted plus the distance to the end of the vector.
...
"